### ISSUE TITLE
i2pd silently exits with errorlevel=1

### ISSUE THREAD

**Issue by hypnosis-i2p** [PurpleI2P/i2pd#1225]
must report smth somewhere

```
root@umbrella:~/.i2pd# rm /var/log/i2pd -rf
root@umbrella:~/.i2pd# i2pd
root@umbrella:~/.i2pd# echo $?
1
root@umbrella:~/.i2pd# ulimit -n
8000
root@umbrella:~/.i2pd# ls /var/log/i2pd
ls: невозможно получить доступ к '/var/log/i2pd': Нет такого файла или каталога
root@umbrella:~/.i2pd# which i2pd
/usr/sbin/i2pd
root@umbrella:~/.i2pd# ls /usr/sbin/i2pd -l
-rwxr-xr-x 1 root root 2398784 янв 31  2018 /usr/sbin/i2pd
root@umbrella:~/.i2pd# df . -h
Файл.система   Размер Использовано  Дост Использовано% Cмонтировано в
/dev/sda7         73G          63G  6,5G           91% /
root@umbrella:~/.i2pd# 
```

**Comment by hypnosis-i2p** [comment]
root@umbrella:~/.i2pd# ps auxwww|grep i2pd
root     10998  0.4  0.3 912036 12160 pts/1    Sl+  авг15   2:38 i2pd
root     16017  0.0  0.0  21316  1016 pts/3    S+   07:15   0:00 grep --color=auto i2pd
root@umbrella:~/.i2pd#

**Comment by LLE8** [comment]
Было? https://github.com/PurpleI2P/i2pd/issues/1153
Там похоже в самом начале LogPrint() еще не живой, может использовать syslog() вместо него местами?

**Comment by hypnosis-i2p** [comment]
нет, тут проблема другая. тут имеется существующий процесс и2пд, и второй процесс и2пд вылетает с ерркодом 1 молча

**Comment by LLE8** [comment]
Конкретная ситуация возможно другая, но подозреваю, что он пытается что-то сказать при помощи LogPrint()

**Comment by LLE8** [comment]
А может быть так, что второй экземпляр i2pd не смог получить доступ к `~/.i2pd/i2pd.pid` ?

**Comment by orignal** [comment]
так оно и есть. если стоит блокировка на pid файле то не запустится.
Это в UnixDaemon.cpp

**Comment by LLE8** [comment]
И в этом самом UnixDaemon.cpp LogPrint() неживой. Тоже долго не мог понять почему оно у меня в OpenWrt молча не запускалось, потом добавил рядом с вызовами LogPrint() вызовы syslog() и сразу в логах прочитал, где у меня ошибка в openwrt-шном пакете.

**Comment by orignal** [comment]
В этот момент он вроде бы должен писать в stdout. Я погляжу

**Comment by orignal** [comment]
Несколько экземляров запускать можно из разных директорий

**Comment by LLE8** [comment]
Кажется у меня в OpenWrt он в stdout не писал.

**Comment by LLE8** [comment]
Даже если можно запускать несколько экземпляров, то зачем это нужно?

**Comment by radfish** [comment]
On Fri, Aug 17, 2018 at 04:09:39AM -0700, orignal wrote:
> В этот момент он вроде бы должен писать в stdout. Я погляжу

я тоже заметил, что если указать --pidfile в несуществующей директории
то на стдаут ничего не пишет, умирает с кодом 1 и всё.
